### PR TITLE
[ASCII-2568] Remove Meter function from telemetry component

### PR DESCRIPTION
**Pull request by GustavoCaso**
### What does this PR do?

This PR removes the `Meter` function from the telemetry component. 
Back in 2022 when the RFC for components was created we said that we would use the Otel MeterProvider for instrument internal components. https://github.com/DataDog/architecture/blob/master/rfcs/agent-component-architecture/rfc.md#telemetry

As of today, we haven't found any internal usage. 

Removing means we no longer add import many `go.opentelemetry.io` libraries into the different Agent binaries. 

If in the future we find internal usage, we can add it 😄 

### Motivation

removed unused function and libraries from the codebase.

### CODE REVIEW

File: comp/core/telemetry/go.mod [hunk 1]
```diff
@@ -25,19 +22,15 @@ require (
 	github.com/beorn7/perks v1.0.1 // indirect
 	github.com/cespare/xxhash/v2 v2.2.0 // indirect
 	github.com/davecgh/go-spew v1.1.1 // indirect
-	github.com/go-logr/logr v1.4.1 // indirect
-	github.com/go-logr/stdr v1.2.2 // indirect
 	github.com/golang/protobuf v1.5.3 // indirect
+	github.com/google/go-cmp v0.6.0 // indirect
```
Comment (pgimalac) [inline comp/core/telemetry/go.mod:26]: Why is this added ?

**Review by pgimalac — APPROVED** [review]
LGTM !

File: comp/core/telemetry/go.mod (hunk 1 above)
Comment (GustavoCaso) [reply comp/core/telemetry/go.mod:26]: I have no idea to be honest. I just run `inv -e tidy`

**Review by mx-psi — APPROVED** [review]

**Review by hush-hush — APPROVED** [review]

**Review by vickenty — APPROVED** [review]

**Review by kkhor-datadog — APPROVED** [review]

**Review by truthbk — APPROVED** [review]
:shipit: you might want to check with @gh123man on Monday (if he's available) if there's any issues with reworking the telemetry interface. Like you said, we don't seem to have any internal use for it, and for the most part, if this is compiling and CI is green I think we're good, but worth having that conversation (it's OK to discuss post-merge).

**Review by jszwedko — APPROVED** [review]

**Review by coignetp — APPROVED** [review]

**Review by ichinaski — APPROVED** [review]
